validator/src/server.rs

```diff
@@ -0,0 +1,463 @@
+//! Validator gRPC high-level server implementation
+
+use std::collections::HashMap;
+
+use tendermint::{
+    block, chain, consensus,
+    proposal::{SignProposalRequest, SignedProposalResponse},
+    public_key::{PubKeyRequest, PubKeyResponse},
+    vote::{SignVoteRequest, SignedVoteResponse},
+    PublicKey, Signature,
+};
+use tendermint_proto::privval::{
+    priv_validator_api_server::{PrivValidatorApi, PrivValidatorApiServer},
+    PubKeyRequest as RawPubKeyRequest, PubKeyResponse as RawPubKeyResponse, RemoteSignerError,
+    SignProposalRequest as RawSignProposalRequest, SignVoteRequest as RawSignVoteRequest,
+    SignedProposalResponse as RawSignedProposalResponse,
+    SignedVoteResponse as RawSignedVoteResponse,
+};
+use tokio::{net::UnixListener, sync::Mutex, time::Instant};
+use tokio_stream::wrappers::UnixListenerStream;
+use tonic::{transport::Server, Request, Response, Status};
+use tracing::{debug, error, info};
+
+use crate::{
+    config::{BasicServerConfig, GrpcSocket},
+    signer::{display_validator_info, SignerProvider},
+    state::ValidatorStateProvider,
+};
+
+/// Validator gRPC high-level server implementation
+#[derive(Debug)]
+pub struct KMSServer<S: SignerProvider, VS: ValidatorStateProvider> {
```

Comment:
Per the [Rust API naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html):

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn

I would suggest the following capitalization instead:

```suggestion
pub struct KmsServer<S: SignerProvider, VS: ValidatorStateProvider> {
```

Reply:
Also if the crate is renamed to `tendermint-privval`, this should probably be called `PrivvalService`